crates/nu-command/src/system/run_external.rs

```diff
@@ -190,15 +204,35 @@ impl<'call, 'contex> ExternalCommand<'call, 'contex> {
                         span: Span::unknown(),
                     }
                 } else {
+                    self.context
+                        .stack
+                        .add_env_var("LAST_EXIT_CODE".to_string(), 0.to_string());
                     Value::nothing()
                 };
 
                 match child.wait() {
```

Comment:
ok. are you saying it's this one instance on line 207 or all of the lines that work this way? it looks like in nushell all of this code would be in a thread::spawn, which, if i remember correctly, makes the scope inaccessible because it's not static and therefore cannot be updated easily. something like that.

Reply:
you've confused me now because i didn't change line 213. that child.wait() has always been there.

Reply:
I know. I mentioned that it wasn't your code hoping you'd see that. Makes me think we should fix the underlying limitation first and then add on the status checking. Doing it in the reverse order might make it harder to fix.